Migrate airflow-kedro guide (#1078)
astronomer/docs: guides/airflow-kedro.md

---
title: "Deploy Kedro Pipelines to Apache Airflow"

> sidebar_label: "Deploy Kedro Pipelines to Apache Airflow"

Reviewer: ```suggestion
sidebar_label: "Kedro"
```

description: "Use the kedro-airflow plugin to change your Kedro pipelines into Apache Airflow DAGs and deploy them to a production environment."
id: airflow-kedro

> tags: ["Plugins", "Integrations"]

Reviewer: ```suggestion
```

---

[Kedro](https://github.com/quantumblacklabs/kedro) is an open-source Python framework for creating reproducible, maintainable, and modular data science code. It borrows concepts from software engineering and applies them to machine learning code.

While Kedro is an excellent option for data engineers and data scientists looking to author their data pipelines and projects with software engineering practices, it can extend even further to integrate with [Apache Airflow](https://airflow.apache.org) for distributed scheduling and execution of the resultant pipelines.

> ## The plugin

Reviewer: ```suggestion
## Airflow Kedro plugin
```

> In close partnership with the team at Kedro, Astronomer recently extended the [`kedro-airflow`](https://github.com/quantumblacklabs/kedro-airflow) plugin to accommodate a significantly improved developer experience. With this plugin, you can translate your Kedro pipeline into a clean, legible, and well-structured Apache Airflow DAG with one simple command:

Reviewer: ```suggestion
In close partnership with the team at Kedro, Astronomer extended the [`kedro-airflow`](https://github.com/quantumblacklabs/kedro-airflow) plugin to accommodate a significantly improved developer experience. With this plugin, you can translate your Kedro pipeline into a clean, legible, and well-structured Apache Airflow DAG with one simple command:
```

```bash
kedro airflow create
```

This makes for a clean experience for anyone looking to deploy their Kedro pipelines to a distributed scheduler for workflow orchestration.

### Prerequisites

> To use the plugin, you'll need the following running on your machine or a fresh virtual environment:
> 
> - [The Astro CLI](https://docs.astronomer.io/astro/install-cli)
> - [Kedro](https://github.com/quantumblacklabs/kedro)
> - [The `kedro-airflow` Plugin](https://github.com/quantumblacklabs/kedro-airflow)
> - [Docker](https://docs.docker.com/docker-for-mac/install/)

Reviewer: ```suggestion
- [The Astro CLI](https://docs.astronomer.io/astro/install-cli).
- [Kedro](https://github.com/quantumblacklabs/kedro).
- [The `kedro-airflow` Plugin](https://github.com/quantumblacklabs/kedro-airflow).
- [Docker](https://docs.docker.com/docker-for-mac/install/).
```

## Try it out

> Astronomer added some functionality to the plugin that makes for a great integration with Astro. To give it a try, you'll use the `astro-airflow-iris` starter that's included in the Kedro project. You'll spin up a fresh Kedro project and run your pipelines as DAGs in a local Airflow environment.

Reviewer: ```suggestion
To try out Astronomer's integration with Airflow, use the `astro-airflow-iris` starter project that includes some simple Kedro pipelines. You'll generate a starter Kedro project, convert your pipelines to Airflow DAGs, and run these DAGs in a local Airflow environment.
```

### Create an Astro-Kedro project

1. Run `kedro new --starter astro-airflow-iris` to build your starter directory.
2. Run `cd <kedro-project-directory>`.
3. Run `kedro install`.
4. Run `kedro package`.
   

> ### Prepare and run the project in Astro

Reviewer: ```suggestion
### Prepare and run the project using the Astro CLI
```

1. Run `cp src/dist/*.whl ./`.
2. Run `kedro catalog create --pipeline=__default__`.
3. Edit your `conf/base/catalog/__default__.yml` and configure datasets to be persisted. For example:

    ```yaml
    example_train_x:
        type: pickle.PickleDataSet
        filepath: data/05_model_input/example_train_x.pkl
    example_train_y:
        type: pickle.PickleDataSet
        filepath: data/05_model_input/example_train_y.pkl
    example_test_x:
        type: pickle.PickleDataSet
        filepath: data/05_model_input/example_test_x.pkl
    example_test_y:
        type: pickle.PickleDataSet
        filepath: data/05_model_input/example_test_y.pkl
    example_model:
        type: pickle.PickleDataSet
        filepath: data/06_models/example_model.pkl
    example_predictions:
        type: pickle.PickleDataSet
        filepath: data/07_model_output/example_predictions.pkl`
    ```

> 4. Make sure you have the `kedro-airflow` plugin installed, then run `pip install kedro-airflow`.

Reviewer: ```suggestion
4. Make sure you have the `kedro-airflow` plugin installed using `pip install kedro-airflow`.
```

> 5. Run `kedro airflow create -t dags/`.

Reviewer: Do you run this command from your Astro project or Kedro project? Do we need to specify a relative path to the `DAGs` directory for the Astro project?

Also, I notice we don't include any of the Dockerfile steps mentioned here: https://kedro.readthedocs.io/en/stable/deployment/airflow_astronomer.html#step-2-package-the-kedro-pipeline-as-an-astronomer-compliant-docker-image. Is that because the `iris` project does this for us, or do we need to configure our Astro project more?

Author: @kentdanas can you address Jake's questions? I have merged this PR, but the tutorial can be updated if changes are required.

Reviewer 2: Your Astro project and your Kedro project are the same in this case. In our guide we direct folks to run `kedro new --starter astro-airflow-iris` first, which is actually bundled with our CLI and runs an `astro dev init` as part of that. So for your second question, yes, the `iris` project does all that other Dockerfile stuff for us (they mention that very briefly in the final note at the end of their tutorial).

I just ran this again and noted that they're still using an older version of our CLI (it pulls an AC 1.10.12 image :cringe:), so at some point we should definitely update this (and probably make it more descriptive). But I think this would likely require working with the Kedro team, so I think we can take care of it post-migration.

Author: @jwitz it appears your questions have been addressed. Can we move this PR forward?

Author: @jwitz can you unblock this PR? It's been sitting for almost two weeks. Thanks!

6. Make sure you have the Astro CLI installed and have Docker running on your machine, then run `astro dev start` to fire up a local Airflow instance and visualize your DAGs.

> Astronomer is proud to partner with Kedro to bring this plugin experience into the world and look forward to extending it to improve the developer experience even more. [Get in touch](https://astronomer.io/contact) if you'd like to talk to us about how you use Kedro and Airflow together!

Reviewer: ```suggestion
```
